src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -1065,10 +1065,10 @@ End Sub
         public async Task CA1707_ClassNameHasUnderscores_Rename()
         {
             await VerifyCS.VerifyCodeFixAsync(@"
-public class [|My_class|]
+public class My_class
 {
 }
-", @"
+", GetCA1707CSharpResultAt(2, 14, SymbolKind.NamedType, "My_class"), @"
```

Comment:
I would instead recommend something like below so we can avoid hard coding magic column/line numbers: https://github.com/dotnet/roslyn-analyzers/blob/03d8e7dbfc82a440aa6b08792693c1a0ea89fe64/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlagTests.cs#L17-L36

Reply:
@mavasani Thanks.
Does the "0" after `{|#` represents the index of the specific reported diagnostic?

Reply:
I got a test failure when I tried to it that way.

![image](https://user-images.githubusercontent.com/31348972/83638697-d46a1700-a5a9-11ea-8183-3c1297cae61e.png)

I've used `#2` because the `TypeRule` has index 2  in the analyzer.
I've tried also `#0` and the test also failed.

@mavasani

Reply:
No, 0 is just a constant indicating tagged location, so `WithLocation(0)` works. I believe you need:
```csharp
VerifyCS.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.TypeRule).WithLocation(0).WithArguments(...));
```